crates/transaction-pool/src/pool/pending.rs

```diff
@@ -129,39 +142,52 @@ impl<T: TransactionOrdering> PendingPool<T> {
         self.by_id.values().map(|tx| tx.transaction.transaction.clone())
     }
 
-    /// Removes all transactions and their dependent transaction from the subpool that no longer
-    /// satisfy the given basefee (`tx.fee < basefee`)
+    /// Updates the pool with the new base fee. Reorders transactions by new priorities. Removes
+    /// from the subpool all transactions and their dependents that no longer satisfy the given
+    /// base fee (`tx.fee < base_fee`).
     ///
     /// Note: the transactions are not returned in a particular order.
-    pub(crate) fn enforce_basefee(
+    ///
+    /// # Returns
+    ///
+    /// Removed transactions that no longer satisfy the base fee.
+    pub(crate) fn update_base_fee(
         &mut self,
-        basefee: u64,
+        base_fee: u64,
     ) -> Vec<Arc<ValidPoolTransaction<T::Transaction>>> {
-        let mut to_remove = Vec::new();
-
-        {
-            let mut iter = self.by_id.iter().peekable();
-            while let Some((id, tx)) = iter.next() {
-                if tx.transaction.transaction.max_fee_per_gas() < basefee as u128 {
-                    // this transaction no longer satisfies the basefee: remove it and all its
-                    // descendants
-                    to_remove.push(*id);
-                    'this: while let Some((peek, _)) = iter.peek() {
-                        if peek.sender != id.sender {
-                            break 'this
-                        }
-                        to_remove.push(**peek);
-                        iter.next();
+        // Create a collection for removed txs.
+        let mut removed = Vec::new();
+
+        // Drain and iterate over all transactions.
+        let mut transactions_iter = self.clear_transactions().into_iter().peekable();
```

Comment:
why do we clear all lists?

we only need to rebuild the sorted list and only remove those that should be removed from the entire set?

Reply:
we need to update two sorted lists